cmd/metal-api/internal/metal/machine_test.go

```diff
@@ -195,3 +195,125 @@ func TestMachineNetwork_NetworkType(t *testing.T) {
 }
 
 // TODO: Write tests for machine allocation
+
+func TestEgressRule_Validate(t *testing.T) {
+	tests := []struct {
+		name     string
+		Protocol Protocol
+		Ports    []int
+		To       []string
+		Comment  string
+		wantErr  bool
+	}{
+		{
+			name:     "valid egress rule",
+			Protocol: ProtocolTCP,
+			Ports:    []int{1, 2, 3},
+			To:       []string{"1.2.3.0/24", "2.3.4.5/32"},
+			Comment:  "allow apt update",
+		},
+		{
+			name:     "wrong protocol",
+			Protocol: Protocol("sctp"),
+			Ports:    []int{1, 2, 3},
+			To:       []string{"1.2.3.0/24", "2.3.4.5/32"},
+			Comment:  "allow apt update",
+			wantErr:  true,
```

Comment:
Testing errors with bool is problematic because it's not clear what produced the error and maybe it fails because something different. 😅

Reply:
done